### PR TITLE
Add a few type annotations

### PR DESCRIPTION
**Pull request by tkdchen**
Signed-off-by: Chenxiong Qi <cqi@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/util.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 This software may be modified and distributed under the terms
 of the BSD license. See the LICENSE file for details.
 """
-
```
Comment (MartinBasti) [inline atomic_reactor/util.py:8]: Newline should stay, but I need little bit of explanation why `_hashlib` ?

Comment (tkdchen) [reply atomic_reactor/util.py:8]: I wanted to describe the type of return value from hashlib methods like `hashlib.md5()` or `hashlib.sha256()`. `type(returned object)` returns `_hashlib.HASH`.

**Review by lkolacek — APPROVED** [review]

File: atomic_reactor/util.py (hunk 1 above)
Comment (tkdchen) [reply atomic_reactor/util.py:8]: The newline is back.

File: atomic_reactor/plugins/pre_pull_base_image.py [hunk 2]
```diff
@@ -176,7 +178,7 @@ def _ensure_image_registry(self, image):
             raise RuntimeError("Shouldn't happen, images should have already "
                                "registry set in dockerfile_images")
 
-    def _pull_and_tag_image(self, image, nonce):
+    def _pull_and_tag_image(self, image: ImageName, nonce: str) -> Optional[ImageName]:
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_pull_base_image.py:181]: I don't think the return value is Optional, the only `return` returns an ImageName and if the return doesn't happen then you get a RuntimeError

File: atomic_reactor/util.py [hunk 3]
```diff
@@ -793,7 +798,7 @@ def get_manifest_digests(self,
 
         return ManifestDigest(**digests)
 
-    def get_manifest_list(self, image):
+    def get_manifest_list(self, image: ImageName) -> requests.Response:
```
Comment (chmeliik) [inline atomic_reactor/util.py:801]: The return value is Optional I think

File: atomic_reactor/util.py [hunk 4]
```diff
@@ -1671,41 +1680,41 @@ def base_image_key(self):
...
-    def organization(self):
+    def organization(self) -> Optional[str]:
         return self._organization
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._pullable_parents)
 
-    def __setitem__(self, key, item):
+    def __setitem__(self, key: Union[str, ImageName], item: str):
```
Comment (chmeliik) [inline atomic_reactor/util.py:1717]: Technically `item` can be an ImageName too, not sure if it's used that way though (`ImageName.parse()` works on ImageName objects too)

File: atomic_reactor/plugins/pre_pull_base_image.py [hunk 5]
```diff
@@ -124,7 +126,7 @@ def _get_image_with_digest(self, image):
         new_image = '{}@sha256:{}'.format(image_name, digest)
         return ImageName.parse(new_image)
 
-    def _store_manifest_digest(self, image, use_original_tag):
+    def _store_manifest_digest(self, image: ImageName, use_original_tag: bool) -> None:
```
Comment (ben-alkov) [inline atomic_reactor/plugins/pre_pull_base_image.py:129]: Redundant? Functions return `None` by default...

Comment (tkdchen) [reply atomic_reactor/plugins/pre_pull_base_image.py:129]: This is really a good question. I think there should be many aspects of the consideration to answer this question. In short, explicit would be better than implicit to describe the behavior of a function. Annotating None here means this method does not return anything.

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by ben-alkov — APPROVED** [review]

**Pull request merged**